web/src/main/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilter.java

```diff
@@ -237,29 +239,19 @@ public WebRequestTrackingFilter() {
     }
 
     private synchronized void initialize(FilterConfig filterConfig) {
+
         StopWatch sw = StopWatch.createStarted();
-        try {
 
-            //if agent is not installed (jar not loaded), can skip the entire registration process
-            try {
-                AgentConnector test = AgentConnector.INSTANCE;
-            } catch (ThreadDeath td) {
-                throw td;
-            } catch (Throwable t) {
-                try {
-                    InternalLogger.INSTANCE.info("Agent was not found. Skipping the agent registration");
-                    return;
-                } catch (ThreadDeath td) {
-                    throw td;
-                } catch (Throwable t2) {
-                    // chomp
-                }
-            }
+        //If Agent Jar is not present in the class path skip the process
+        if (!CommonUtils.isClassPresentOnClassPath(AGENT_LOCATOR_INTERFACE_NAME,
```

Comment:
Since this operation is being timed, let's get 5 samples of this running to a successful and a failed state. We should understand the perf impact of this change.

Reply:
Cool, that means with this app we see it perform ~23.3% slower in the failure case (agent not found) and ~1.77% faster in the success case (agent found). The slowdown is significant, but since that case is on the order of microseconds, I think this is reasonable and we shouldn't see significant impact on slower hardware.

We should do the same test on a "hello world" app just to confirm the assumption that more classes means more time in `Class.forName`. That'll give us a bit more insight into how this will scale.

@grlima, what do you think?